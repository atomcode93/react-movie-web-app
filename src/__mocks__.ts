export const initialState = {
  auth: {
    error: false,
    isAuthenticated: false,
    user: null,
  },
  favorites: {
    error: false,
    favoritedIds: [],
    favoritedMovies: null,
  },
  filters: {
    order: {
      label: 'Descending',
      value: 'desc',
    },
    rating: {
      max: 10,
      min: 5,
    },
    runtime: {
      max: 250,
      min: 45,
    },
    sort_by: {
      label: 'Rating',
      value: 'vote_average',
    },
    year: 2019,
  },
  modals: {
    args: null,
    open: false,
    type: null,
  },
  movies: {
    error: false,
    favorites: [],
    loading: false,
    movie: null,
    movies: null,
    popularMovies: null,
    soonMovies: null,
    topRatedMovies: null,
    watchLater: null,
  },
  router: {
    action: 'POP',
    location: {
      hash: '',
      pathname: '/',
      search: '',
      state: undefined,
    },
  },
  watchLater: {
    error: false,
    watchLaterIds: [],
    watchLaterMovies: null,
  },
};
