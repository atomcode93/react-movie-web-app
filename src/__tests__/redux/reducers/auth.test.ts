import {
  actions as authActions,
  reducer as authReducers,
} from 'redux/ducks/auth.duck';
import { initialState } from '__mocks__';

const authInitialState = initialState.auth;

describe('Auth Reducer', () => {
  test('should setup default modals state', () => {
    const state = authReducers(undefined, { type: '@@INIT' });
    expect(state).toEqual(authInitialState);
  });
  test('should set user', () => {
    const action = authActions.setUserSuccess({
      type: 'SET_USER_SUCCESS',
      payload: {
        user: {},
      },
    });
    const state = authReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
  test('should set initial state on authLogout', () => {
    const action = authActions.authLogout();
    const state = authReducers(undefined, action);
    expect(state).toMatchSnapshot();
  });
});
