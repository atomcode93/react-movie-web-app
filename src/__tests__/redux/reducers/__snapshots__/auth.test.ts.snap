// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Reducer should set initial state on authLogout 1`] = `
Object {
  "error": false,
  "isAuthenticated": false,
  "user": null,
}
`;

exports[`Auth Reducer should set user 1`] = `
Object {
  "error": false,
  "isAuthenticated": true,
  "user": Object {
    "payload": Object {
      "user": Object {},
    },
    "type": "SET_USER_SUCCESS",
  },
}
`;
